src/controllers/llmo.js

```diff
@@ -196,13 +205,98 @@ function LlmoController() {
     return ok(config.getLlmoConfig().questions);
   };
 
+  // Handles requests to the LLMO customer intent endpoint, returns customer intent array
+  const getLlmoCustomerIntent = async (context) => {
+    const { llmoConfig } = await getSiteAndValidateLlmo(context);
+    return ok(llmoConfig.customerIntent || []);
+  };
+
+  // Handles requests to the LLMO customer intent endpoint, adds new customer intent items
+  const addLlmoCustomerIntent = async (context) => {
+    const { log } = context;
+    const { site, config } = await getSiteAndValidateLlmo(context);
+
+    const newCustomerIntent = context.data;
+    if (!newCustomerIntent || !Array.isArray(newCustomerIntent)) {
+      return badRequest('Customer intent must be provided as an array');
+    }
+
+    // Validate structure of each customer intent item
+    for (const intent of newCustomerIntent) {
+      if (!intent.key || !intent.value) {
+        return badRequest('Each customer intent item must have both key and value properties');
+      }
+      if (typeof intent.key !== 'string' || typeof intent.value !== 'string') {
+        return badRequest('Customer intent key and value must be strings');
+      }
+    }
+
+    config.addLlmoCustomerIntent(newCustomerIntent);
+    await saveSiteConfig(site, config, log, 'adding customer intent');
+
+    // return the updated llmoConfig customer intent
+    return ok(config.getLlmoConfig().customerIntent || []);
+  };
+
+  // Handles requests to the LLMO customer intent endpoint, removes a customer intent item
+  const removeLlmoCustomerIntent = async (context) => {
+    const { log } = context;
+    const { intentKey } = context.params;
+    const { site, config } = await getSiteAndValidateLlmo(context);
+
+    validateCustomerIntentKey(config, intentKey);
+
+    // remove the customer intent using the config method
+    config.removeLlmoCustomerIntent(intentKey);
+
+    await saveSiteConfig(site, config, log, 'removing customer intent');
+
+    // return the updated llmoConfig customer intent
+    return ok(config.getLlmoConfig().customerIntent || []);
+  };
+
+  // Handles requests to the LLMO customer intent endpoint, updates a customer intent item
+  const patchLlmoCustomerIntent = async (context) => {
+    const { log } = context;
+    const { intentKey } = context.params;
+    const { data } = context;
+    const { site, config } = await getSiteAndValidateLlmo(context);
+
+    validateCustomerIntentKey(config, intentKey);
+
+    // Validate the update data
+    if (!data || typeof data !== 'object') {
```

Comment:
you can use the `isObject` utility function available in spacecat-shared-utils

Reply:
https://github.com/adobe/spacecat-api-service/pull/1141/commits/bb0d0ce55f2b63d23d8a922e68998b4b7a877743